sdk/tables/data-tables/src/serialization.ts

```diff
@@ -157,23 +157,35 @@ export function deserialize<T extends object = Record<string, any>>(
   return deserialized;
 }
 
-function inferTypedObject(propertyName: string, value: number | string) {
+function inferTypedObject(propertyName: string, value: number | string | boolean) {
   // We need to skip service metadata fields such as partitionKey and rowKey and use the same value returned by the service
   if (propertyCaseMap.has(propertyName)) {
     return value;
   }
 
-  return typeof value === "string" ? { value, type: "String" } : getTypedNumber(value);
+  const valuStr = String(value);
+
+  switch (typeof value) {
+    case "boolean":
+      return { value: valuStr, type: "Boolean" };
```

Comment:
Feels like this is the only case we need it for:

```suggestion
      return { value: String(value), type: "Boolean" };
```